sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/MicrosoftBotIdentifier.java

```diff
@@ -0,0 +1,135 @@
+     * @param isResourceAccountConfigured Set this to true if the bot is tenantized.
+     * It is false if the bot is global and no resource account is configured.
+     * @param cloudEnvironment the cloud environment in which this identifier is created.
+     * @throws IllegalArgumentException thrown if botId parameter fail the validation.
+     */
+    public MicrosoftBotIdentifier(String botId,
+                                  boolean isResourceAccountConfigured,
+                                  CommunicationCloudEnvironment cloudEnvironment) {
+        if (botId == null || botId.trim().length() == 0) {
+            throw new IllegalArgumentException("The initialization parameter [botId] cannot be null or empty.");
+        }
+        this.botId = botId;
+        this.cloudEnvironment = cloudEnvironment;
+        this.isResourceAccountConfigured = isResourceAccountConfigured;
+        generateRawId();
+    }
+
+    /**
+     * Creates a MicrosoftBotIdentifier object
+     *
+     * @param botId The unique Microsoft app ID for the bot as registered with the Bot Framework.
+     * @param isResourceAccountConfigured Set this to true if the bot is tenantized.
+     * It is false if the bot is global and no resource account is configured.
+     * @throws IllegalArgumentException thrown if botId parameter fail the validation.
+     */
+    public MicrosoftBotIdentifier(String botId, boolean isResourceAccountConfigured) {
+        this(botId, isResourceAccountConfigured, CommunicationCloudEnvironment.PUBLIC);
+    }
+
+    /**
+     * Creates a MicrosoftBotIdentifier object
+     *
+     * @param botId The unique Microsoft app ID for the bot as registered with the Bot Framework.
+     * @throws IllegalArgumentException thrown if botId parameter fail the validation.
+     */
+    public MicrosoftBotIdentifier(String botId) {
+        this(botId, true, CommunicationCloudEnvironment.PUBLIC);
+    }
+
+    /**
+     * Get the Microsoft app ID for the bot.
+     * @return microsoftBotId Id of the Microsoft app ID for the bot.
+     */
+    public String getBotId() {
+        return this.botId;
+    }
+
+    /**
+     * @return True if the bot is tenantized and false if the bot is global and no resource account is configured.
+     */
+    public boolean isResourceAccountConfigured() {
+        return this.isResourceAccountConfigured;
+    }
+
+    /**
+     * Get cloud environment of the Microsoft bot identifier.
+     * @return cloud environment in which this identifier is created.
+     */
+    public CommunicationCloudEnvironment getCloudEnvironment() {
+        return cloudEnvironment;
+    }
+
+    /**
+     * Set full id of the identifier.
+     * RawId is the encoded format for identifiers to store in databases or as stable keys in general.
+     *
+     * @param rawId full id of the identifier.
+     * @return MicrosoftBotIdentifier object itself.
+     */
+    @Override
+    public MicrosoftBotIdentifier setRawId(String rawId) {
```

Comment:
we could possibly avoid the whole `rawIdSet` fiddling if we didn't expose the `setRawId` method publicly. do we need to do that?

Reply:
so as the setRawId is available for the public already from the parent class of the bot identifier, so I think we can not avoid from it